optaplanner-examples/src/main/java/org/optaplanner/examples/cloudbalancing/optional/benchmark/CloudBalancingBenchmarkHelloWorld.java

```diff
@@ -16,19 +16,68 @@
 
 package org.optaplanner.examples.cloudbalancing.optional.benchmark;
 
+import java.util.Arrays;
+import java.util.List;
+
 import org.optaplanner.benchmark.api.PlannerBenchmark;
 import org.optaplanner.benchmark.api.PlannerBenchmarkFactory;
+import org.optaplanner.benchmark.impl.aggregator.swingui.BenchmarkAggregatorFrame;
+import org.optaplanner.core.api.solver.SolverFactory;
+import org.optaplanner.examples.cloudbalancing.app.CloudBalancingHelloWorld;
+import org.optaplanner.examples.cloudbalancing.domain.CloudBalance;
+import org.optaplanner.examples.cloudbalancing.persistence.CloudBalancingGenerator;
 
+/**
+ * @see CloudBalancingHelloWorld
+ */
 public class CloudBalancingBenchmarkHelloWorld {
 
     public static void main(String[] args) {
+        List<String> argList = Arrays.asList(args);
+        boolean advanced = argList.contains("--advanced");
+        if (!advanced) {
+            runBasicBenchmark();
+        } else {
+            boolean aggregator = argList.contains("--aggregator");
+            runAdvancedBenchmark(aggregator);
+        }
```

Comment:
Shouldn't this behavior be somehow shared across all the individual benchmark apps that can be found in the examples?

Reply:
This is the only hello world class for a benchmark - all the BenchmarkApp classes use the "examples common framework".